semantic-python/src/Language/Python.hs

```diff
@@ -1,19 +1,222 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DuplicateRecordFields      #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE PatternSynonyms            #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeOperators              #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
```

Comment:
IMO we should definitely declare a new, internal-only typeclass. We gain little to nothing from sharing the class interface given that `Term` mediates the public interface to the language, and orphan instances are a code smell at best.

If we want to share mechanisms, we can write them alongside-but-not-dependent-upon the typeclass, à la the `yield` & `runTagging` machinery in `Tags.Tagging.Precise`—neither uses `toTags`, both are used to implement instances of the per-language classes in `Language.*.Tags`.

Reply:
I also feel pretty strongly that these instances don’t belong in this module either way—this module exists primarily to vend `Term`. Let’s put them in a new module to avoid stepping on each others’ toes as we’re working on disjoint functionality for the same languages.

Reply:
This has all been instituted.